### ISSUE TITLE
Allow multiple instances of the -command flag 

### ISSUE THREAD

**Issue by slavistan** [gokcehan/lf#551]
**tl;dr** Currently, simultaneous usage of multiple *-command* flags is not supported. `lf -command foo -command bar` will only execute the latter. Support for multiple *-command* instances would greatly improve `lf`'s interface for shell scripts by allowing to freely combine (mutually compatible) behaviors and keybindings implemented in disparate configuration files through multiple usages of the *-command* flag, e.g. `lf -command "source lfrc-foo" -command "source lfrc-bar"`.

`lf -command ...` appends commands to `lf`'s startup sequence which greatly facilitates scripting. More specifically, `lf -command "source /path/to/lfrc-addendum"` allows scripts to add usage-specific behavior and keybindings without polluting one's default *lfrc*. This is a very versatile usage pattern which permits one to wrap `lf` in a thin script to [browse youtube](https://github.com/slavistan/lf-gadgets/tree/master/lf-yt), to generate [file previews with ueberzug](https://github.com/slavistan/lf-gadgets/tree/master/lf-ueberzug) or to [interface `lf` with one's interactive shell](https://github.com/slavistan/lf-gadgets/tree/master/lf-shellcd) - with absolutely no requirements to change the default behavior of `lf` or one's *lfrc*.

![](https://raw.githubusercontent.com/slavistan/lf-gadgets/master/lf-yt/demo.gif)

When two or more of such usage-specific behaviors are orthogonal it may be desirable to combine them. For example, image previews and shell integration (e.g. changing the cwd) are independent and can work in unison. The natural way to do this, while maintaining the scripts' independence, is to have one `lf` wrapper call another `lf` wrapper, both of which load their specific configuration snippets via their respective *-command "source ..."* parameter (and pass down their arguments via the shell's `$@`). This resolves to a call to `lf` with **two** *-command* flag instance, one for each wrapper. And here's the problem: Multiple occurences of the *-command* flag currently do not work - only the last instance will be executed. This will break the the outer script since its usage of *-command "source ..."* will be ignored.

Support for multiple instances of *-command* would fix this by facilitating the generic pattern of loading multiple independent configuration snippets (executed in their order of appearance). `lf -command foo -command bar -command baz` should run the three commands in their order of appearance. One of `lf`'s most salient selling points is its scriptability and compatibility with other tools which has seen great developments in the recent weeks (e.g. the better interface for terminal image previews) and I feel that this feature would strike in the same notch.

What do you think?

---

Here's a final, more detailed walkthrough: Consider two functionally compatible wrapper scripts, `lf-foo`, which calls `lf`, and `lf-bar`, which calls `lf-foo`. Each of the wrappers has its own configuration in an rc-file called *lf-foo* and *lf-bar*, respectively.

```
# lf-foo - wrapper around lf
lf -command "source lfrc-foo" "$@"
```

```
# lf-bar - wrapper around lf-foo
lf-foo -command "source lfrc-bar" "$@"
```

A call to `lf-bar` will thus resolve to

```
lf -command "source lfrc-bar" -command "source lfrc-foo" "$@"
```

Due to the missing support of multiple *-command* instances only *lfrc-foo* will be sourced, breaking `lf-bar` due to failing to source *lfrc-bar*.

**Comment by gokcehan** [comment]
@slavistan Does this not work already with `lf -remote 'foo; bar'`?

**Comment by slavistan** [comment]
@gokcehan While multiple **-command**s may indeed be collated if they're all known in advance, this approach fails if the contributions of the individual **-command**s to the final call to `lf` are spread among different scripts - *foo* need not know anything about *bar*. This is precisely the situation which the example attempts to convey. 

If I'm not making sense please let me know. I'm attempting to convey the idea without being overly verbose. There's certainly more detail I can provide. 

#552 seems to do the job perfectly.

**Comment by gokcehan** [comment]
@slavistan Sorry, I realize now you already have the answer to my question in your post. To be honest, I didn't read the bottom half carefully before. #552 seems good to me. I will merge it soon.
